04_verify.sh

```diff
@@ -41,67 +76,94 @@ differs(){
   if [[ "${1}" != "${2}" ]]; then
     echo "OK - ${3}" >&3
     echo "${FAILS}"
+    return 0
   else
     echo "FAIL - ${3}" >&3
     echo "$((FAILS+1))"
+    return 1
   fi
 }
 
 
 check_bm_hosts() {
-    while read -r name address user password mac; do
-      BM_VMNAME="${name//-/_}"
-      # Verify BM host exists
-      echo "$BM_HOSTS" | grep -w "${name}"  > /dev/null
-      FAILS="$(process_status $? "${name} Baremetalhost exist")"
-
-      BM_HOST="$(echo "${BM_HOSTS}" | jq ' .items[] | select(.metadata.name=="'"${name}"'" )')"
-
-      # Verify addresses of the host
-      FAILS="$(equals "$(echo "${BM_HOST}" | jq -r '.spec.bmc.address')" \
-        "${address}" "${name} Baremetalhost address correct")"
-
-      FAILS="$(equals "$(echo "${BM_HOST}" | jq -r '.spec.bootMACAddress')" \
-        "${mac}" "${name} Baremetalhost mac address correct")"
-
-      # Verify BM host status
-      FAILS="$(equals "$(echo "${BM_HOST}" | jq -r '.status.operationalStatus')" \
-        "OK" "${name} Baremetalhost status OK")"
-
-      # Verify credentials exist
-      CRED_NAME="$(echo "${BM_HOST}" | jq -r '.spec.bmc.credentialsName')"
-      CRED_SECRET="$(kubectl get secret "${CRED_NAME}" -n metal3 -o json | jq '.data')"
```

Comment:
in the case we use kinder, I think we may need to export KUBECONFIG
```sh
export KUBECONFIG="kinder get kubeconfig-path"
```

Reply:
ok, I'll add a Kubeconfig variable